<a href="https://colab.research.google.com/github/Ignacio737/PrimerProyectoFisica/blob/master/SeriePandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Serie Pandas
Esta es la segunda serie, en esta quiero que trabajemos con datasets que se obtendrán a partir del github del curso que es: https://github.com/adrishg/PythonML-PrebecariosProteco 

(No les subiré un punto por seguirme porque no estoy tan desesperada por popularidad pero deberían considerarlo)

In [0]:
import pandas as pd

1) Considerando el siguiente set de datos:
           
           a) Crea un dataframe a partir de datos
           b) Despliega los primeros tres datos de dicho dataframe
           c) Despliga los últimos dos datos de dicho dataframe
           d) Una descripción de los datos (count, mean, std, ...) 

In [0]:
datos = {
    'fecha':pd.date_range('20200430', periods=6),
    'calidad':pd.Categorical(['buena', 'muy buena', 'mala', 'buena', 'muy mala', 'buena'],
                                 categories = ['muy mala', 'mala', 'regular', 'buena', 'muy buena'],
                                 ordered = True),
    'asistentes':[10, 25, 12, 3, 23, 43],
    'recomendaciones':['hablar más lento', 'me aburri', 'Derek hacia cosas raras', 'excelente', 'no se entendio nada',
               '¿Y el recreo?']
}

In [0]:
dataFrameDatos=pd.DataFrame(datos)
dataFrameDatos



fecha    calidad  asistentes          recomendaciones
0 2020-04-30      buena          10         hablar más lento
1 2020-05-01  muy buena          25                me aburri
2 2020-05-02       mala          12  Derek hacia cosas raras
3 2020-05-03      buena           3                excelente
4 2020-05-04   muy mala          23      no se entendio nada
5 2020-05-05      buena          43            ¿Y el recreo?

2) Crea una función que pida número de columnas, su nombre y número de datos a ingresar de manera que el usuario pueda crear un dataframe con los datos de su preferencia.

In [0]:
 import pandas as pd 
 def creadorDataFrames():
     datosingresados={}
     print("numero de columnas:")
     numeroColumnas=int(input())
     print("numero de filas:")
     numeroFilas=int(input())
     for i in range(numeroColumnas):
       print("Nombre de",i+1,"columna")
       nombreColumna=input()
       columna=[]
       for j in range(numeroFilas):
           print(j+1,"dato") 
           dato= input()
           columna.append(dato)
       datosingresados[nombreColumna]=columna
dataFrameingresado=pd.DataFrame(datosingresados)
print(dataFrameingresado)

creadorDataFrames()
       


NameError: ignored

3) A partir del siguiente DataSet realiza:

In [0]:
#Liga del github: https://raw.githubusercontent.com/adrishg/PythonML-PrebecariosProteco/master/DataSets/startupsUSA.csv
#Podrán agregar archivos de github y leerlos usando el url raw
data = pd.read_csv('https://raw.githubusercontent.com/adrishg/PythonML-PrebecariosProteco/master/DataSets/startupsUSA.csv')
data.head()

R&D Spend  Administration  Marketing Spend       State     Profit
0  165349.20       136897.80        471784.10    New York  192261.83
1  162597.70       151377.59        443898.53  California  191792.06
2  153441.51       101145.55        407934.54     Florida  191050.39
3  144372.41       118671.85        383199.62    New York  182901.99
4  142107.34        91391.77        366168.42     Florida  166187.94

    a) Crea una función que reciba un lugar y despligue en el número de empresas que residen en dicho lugar

In [0]:
def contarPorLugar(lugar):
    estaAhi = data['State'] == lugar
    agrupadoPorLugar = data[estaAhi]
    print("El numero de empresas en",lugar,"es",len(agrupadoPorLugar))
    print(agrupadoPorLugar)

contarPorLugar('New York')


NameError: ignored

    b) Un data set cuyo orden este dado alfabeticamente por los lugares donde están las empresas

In [0]:
ordenadorPorNombreLugar = data.sort_values('State',ascendin=True)
print(ordenadoPorNombreLugar.head())

NameError: ignored

    c) Una función donde ingreses el index númerico y despliegue dichos datos de manera legible:
    
        Ejemplo: [In]: 0  [Out]: La empresa que reside en New York tiene un gato de Marketing de 471784.10 dólares mientras que sus ganancias son 192261.83 dólares...

In [0]:
def accerderPorIndice(indice):
    resgistroPorIndice = data.iloc[indice]
    print("La empresa que reside en", registroPorIndice['State'],"tiene un
    gasto ")

4) Deberán cargar los datos de iris setosa de dos maneras:

    a) Deberán dirigirse al repositorio del curso: https://github.com/adrishg/PythonML-PrebecariosProteco
    De ahí a la carpeta de DataSets, ahí encontrarán el archivo iris.csv deben darle click para previsualizar el archivo. Ahí encontrarán un botón que dice Raw denle click y copien la liga, ese es el link raw y podrán exportarlo como lo hice en la parte de arriba con los datos de las empresas estadounidenses.
    Desplieguen los datos con un describe para que revisen que se exporto bien.
    
    NOTA: También pueden aprovechar y seguirme en github, saludos.
    
![screenshot iris csv](https://drive.google.com/uc?id=1ytlucQdsbdnWsJmTG61eR7WQ_PBFNmCQ)
    

In [0]:
data = pd.read_csv('https://raw.githubusercontent.com/adrishg/
,!PythonML-PrebecariosProteco/master/DataSets/iris.csv')
data.head()

SyntaxError: ignored

    b) El dataset de Iris es muy conocido en el área ya que está precargado en sklearn.datasets. Pueden obtenerlo a partir de este, tendrán que investigarlo y exportar los datos.

In [0]:
from sklearn.datasets import load_iris


iris = load_iris()
irisDataFrame = pd.DataFrame(iris.dta, columns=iris.feature_names)
irisDataFrame.head()

NameError: ignored

    c) Por último les recomiendo leer acerca de este dataset, lo usaremos bastante en clase.

5) En esta opción eres libre de poner cualquier función que use elementos y funciones de la librería pandas. 
    El requerimiento es que uses iloc, puedes usar alguno de los datasets cargados en esta libreta de jupyter.
